paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -221,53 +221,50 @@ internal abstract class BaseAddPaymentMethodFragment(
         @VisibleForTesting
         internal fun getFormArguments(
             hasCustomer: Boolean,
-            supportedPaymentMethodName: String,
+            supportedPaymentMethod: SupportedPaymentMethod,
             stripeIntent: StripeIntent,
             merchantName: String,
             amount: Amount? = null,
             billingAddress: PaymentSheet.BillingDetails? = null
         ): FormFragmentArguments {
-            // Has effect of setting off session on PIs (not SIs) and also impacts reopening
-            // the card
-            var saveForFutureUseValue = true
-            // This will impact the setting of the off_session on confirm
-            var saveForFutureUseVisible = true
-
-            val supportedPaymentMethod =
-                SupportedPaymentMethod.fromCode(supportedPaymentMethodName)
 
             val isSetupIntent = stripeIntent is SetupIntent
             val isPaymentIntentSetupFutureUsage = (stripeIntent as? PaymentIntent)?.setupFutureUsage
-            if (isSetupIntent ||
-                (isPaymentIntentSetupFutureUsage == StripeIntent.Usage.OnSession) ||
-                (isPaymentIntentSetupFutureUsage == StripeIntent.Usage.OffSession)
-            ) {
-                saveForFutureUseVisible = false
-                saveForFutureUseValue = true
-            } else if (stripeIntent is PaymentIntent &&
-                (!hasCustomer || (supportedPaymentMethod?.requiresMandate == true))
-            ) {
-                // If paymentMethodTypes contains payment method that does not support
-                // save for future should be false and unselected until future fix
-                saveForFutureUseValue = false
-                saveForFutureUseVisible = false
-            } else if (stripeIntent is PaymentIntent) {
-                // If the intent includes any payment method types that don't support save remove
-                // checkbox regardless of the payment method until future fix
-                stripeIntent.paymentMethodTypes.forEach {
-                    if (SupportedPaymentMethod.fromCode(it)
-                        ?.userRequestedConfirmSaveForFutureSupported == false
-                    ) {
-                        saveForFutureUseValue = false
-                        saveForFutureUseVisible = false
+
+            // It impacts reopening the card and together with allowUserInitiatedReuse
+            // will determine if off_session is set on confirm.
+            var isReusable =
+                isSetupIntent ||
+                    (isPaymentIntentSetupFutureUsage == StripeIntent.Usage.OnSession) ||
+                    (isPaymentIntentSetupFutureUsage == StripeIntent.Usage.OffSession)
+            // This will impact the setting of the off_session on confirm
+            var allowUserInitiatedReuse = true
+
+            if (isReusable) {
```

Comment:
Can we rename `isReusable` to `merchantRequiresSave` or something like that?

`isReusable` makes me think that this describes whether the payment method is reusable or not.